fv3core/utils/corners.py

```diff
@@ -54,6 +56,61 @@ def __call__(self, field: FloatField):
         self._copy_corners(field, field)
 
 
+class CopyCornersXY:
+    """
+    Helper-class to copy corners corresponding to the Fortran functions
+    copy_corners_x and copy_corners_y
+    """
+
+    def __init__(
+        self,
+        grid_indexing: GridIndexing,
+        dims: Sequence[str],
+        y_field,
+    ) -> None:
+        """
+        Args:
+            grid_indexing: information about the grid sizing
+            dims: dimensionality of the data to be copied
+            y_field: 3D gt4py storage to use for y-differenceable field
+                (x-differenceable field uses same memory as base field)
+        """
+        origin, domain = grid_indexing.get_origin_domain(
+            dims=dims, halos=(grid_indexing.n_halo, grid_indexing.n_halo)
+        )
+
+        self._y_field = y_field
+
+        ax_offsets = axis_offsets(grid_indexing, origin, domain)
+        self._copy_corners_xy = FrozenStencil(
+            func=copy_corners_xy_stencil_defn,
+            origin=origin,
+            domain=domain,
+            externals={
+                **ax_offsets,
+            },
+        )
+
+    def __call__(self, field: FloatField):
+        """
+        Fills cell quantity field using corners from itself.
+
+        Args:
+            field: field to fill corners
+
+        Returns:
+            x_differenceable: input field, updated so it can be differenced
+                in x-direction
+            y_differenceable: copy of input field which can be differenced
+                in y-direction
```

Comment:
this is just copying the corners, how the resulting field is used after doesn't necessarily have to be tied to x and y being differenceable, though that is the purpose in fvtp2d